INSTALL.md

```diff
@@ -0,0 +1,126 @@
+# Overview
+
+Neural Artistic Style relies on these dependencies:
+
+ * [CUDA Toolkit](https://developer.nvidia.com/cuda-downloads)
+ * [cuDNN](https://developer.nvidia.com/cudnn) v3
+ * [DeepPy](http://github.com/andersbll/deeppy), Deep learning in Python.
+ * [CUDArray](http://github.com/andersbll/cudarray), CUDA-accelerated NumPy.
+ * [Pretrained VGG 19 model](http://www.vlfeat.org/matconvnet/pretrained), choose *imagenet-vgg-verydeep-19*.
+
+The detailed installation steps are explained in the sections below.
+
+Note: This installation in mainly for GNU/Linux distributions.
+
+# Neural Artistic Style
+
+1. Download Neural Artistic Style:
+
+    $ git clone 
+
+# CUDA
+
+Please refer to the [Installation Guides](http://docs.nvidia.com/cuda/index.html#installation-guides) provided by nVidia.
+
+The CUDA toolkit should be installed at `/usr/local/cuda/`.
+
+# cuDNN
+
+1. Download [cuDNN](https://developer.nvidia.com/cudnn) v3.
+1. Extract the tarball file to the CUDA directory:
+
+    $ sudo tar xzf cudnn-7.0-linux-x64-v3.0-prod.tgz -C /usr/loca/cuda
+
+The tarball file consists of libcudnn static and shared object libraries, and
+the library header.
+
+# CUDArray
+
+1. Download CUDArray:
+
+    $ git clone https://github.com/andersbll/cudarray.git
+
+1. Build CUDArray:
+
+Before building CUDArray, please make sure Cython>=0.21 has been installed.  If
+not, you can install Cython via Pip:
+
+    $ pip install --user --upgrade cython
+
+Start to build:
+
+    # Install shared object library
+    $ make
+    $ sudo make install  # install into /usr/local/lib by default
+    $ echo "export LD_LIBRARY_PATH=\"/usr/local/lib:$LD_LIBRARY_PATH\"" >> $HOME/.bashrc
+    $ source $HOME/.bashrc
+
+    # Install Python modules
+    $ sudo python setup.py install
+
+If you get the error messages when executing `make`, that means you might be
+using cuDNN v4 instead of v3 (Issue #36):
+
+    src/nnet/cudnn.cpp:206:5: error: cannot convert ‘const float*’ to ‘cudnnConvolutionBwdFilterAlgo_t’ for argument ‘8’ to ‘cudnnStatus_t cudnnConvolutionBackwardFilter(cudnnHandle_t, const void*, cudnnTensorDescriptor_t, const void*, cudnnTensorDescriptor_t, const void*, cudnnConvolutionDescriptor_t, cudnnConvolutionBwdFilterAlgo_t, void*, size_t, const void*, cudnnFilterDescriptor_t, void*)’
+         ));
+         ^
+    ./include/cudarray/nnet/cudnn.hpp:85:44: note: in definition of macro ‘CUDNN_CHECK’
+     #define CUDNN_CHECK(status) { cudnn_check((status), __FILE__, __LINE__); }
+ 
+    compilation terminated due to -Wfatal-errors.
+    make: *** [src/nnet/cudnn.o] Error 1
+                                                                                                                                         
+https://github.com/andersbll/cudarray/issues/36
+
```

Comment:
Then, the user can add this in their `.bashrc` to avoid issues with bad pathes.
```sh
# CUDArray lib path:
export LD_LIBRARY_PATH="/usr/local/lib" # path to libcudarray.so's folder
```